src/server/player_sao.cpp

```diff
@@ -558,12 +558,40 @@ void PlayerSAO::setMaxSpeedOverride(const v3f &vel)
 
 bool PlayerSAO::checkMovementCheat()
 {
-	if (isAttached() || m_is_singleplayer ||
+	if (m_is_singleplayer ||
 			g_settings->getBool("disable_anticheat")) {
 		m_last_good_position = m_base_position;
 		return false;
 	}
 
+	if (isAttached()) {
+		int parent_id;
+		std::string bone;
+		v3f attachment_pos, attachment_rot;
+		getAttachment(&parent_id, &bone, &attachment_pos, &attachment_rot);
+		UnitSAO *parent = (UnitSAO *)getParent();
+		v3f pos = getBasePosition();
+		f32 radius = sqrt(attachment_pos.X * attachment_pos.X + attachment_pos.Z * attachment_pos.Z);
+		v3f parent_pos = parent->getBasePosition();
+		f32 parent_rot = parent->getRadYawDep() + acos(1 / radius);
+		attachment_pos.X = cos(parent_rot) * radius;
+		attachment_pos.Z = sin(parent_rot) * radius;
+		attachment_pos *= 3;
```

Comment:
What's all this for?

```cpp
		v3f supposed_pos = parent->getBasePosition() + attachment_pos;
		v3f diffvec = getBasePosition() - supposed_pos;
```
Wouldn't this work just as well?

Reply:
`attachment_pos` is applied based on the entity rotation, though the camera is no longer at the player model's head position due to lack of rotation support in the camera code...

Reply:
Maybe it'd be enough to simplify it to:
```c++
	if (getBasePosition().getDistanceFrom(parent->getBasePosition()) > attachment_pos.getLength() * 2.0) {
		/* cheated */
	}
```

Thoughts?

Reply:
Probably better, yes

Reply:
But the problem is: The exact position should be calculated, to the player can be reset to it.

Reply:
Using something like m_last_good_position is no option, as the illegal position change could be done when attaching to the entity.

Reply:
Maybe it is an option to just set the player's pos to the parent's pos, but that would be kinda ugly.

Reply:
> so the player can be reset to it.

Just teleport to the entity's root position. This is a case that should never happen, and the server does not know about bone rotations anyway.

Reply:
Ok

Reply:
It's implemented